### PR TITLE
Remove unused headers

### PR DESCRIPTION
**Pull request by gezalore**
Apologies for the boring PR, the reason I'm PRing this rather than pushing is because it might cause portability issues if I removed something that is only used on some systems, so you might like to comment. For ease of review, here are the unique names removed:

```
 % git show -p | grep ^-# | sort -u
-#include <algorithm>
-#include <cassert>
-#include <cerrno>
-#include <climits>
-#include <cmath>
-#include <cstdarg>
-#include <cstdlib>
-#include <ctime>
-#include <fcntl.h>
-#include <fstream>
-#include <iomanip>
-#include <limits>
-#include <list>
-#include <map>
-#include <set>
-#include <sstream>
-# include <stdint.h>
-#include <sys/stat.h>
-# include <sys/wait.h>   This actually is not removed, I just added a comment.
-#include <tgmath.h>
-#include <utility>
-# include "V3Ast.h"
-#include "V3Changed.h"
-#include "V3Const.h"
-#include "V3Error.h"
-#include "V3File.h"
-#include "V3Global.h"
-#include "V3GraphPathChecker.h"
-#include "V3Inst.h"
-#include "V3LinkCells.h"
-#include "V3Parse.h"
-#include "V3PreShell.h"
-# include "V3Stats.h"
-#include "V3Stats.h"
-#include <vector>
-#include "verilated_intrinsics.h"
-#include "verilatedos.h"
```

The only non C/C++ standard headers are sys/stat.h and fcntl.h but based on where those were used I think it's OK to drop them.

### CODE REVIEW

**Comment by wsnyder** [comment]
These are certainly 95% cruft, but the question is which ones are the other 5% (or is it 0%)?  tgmath.h is probably the most scary as that's quite specific & unusual.

Certainly cleaning up any V3* ones or others in our own sources are fine - if it compiles one place it'll compile everywhere.

To test this we'll need to run on pre-C11, OS-X, Free BSD and Cygwin.  I checked pre-C11, do you have access to others?

**Comment by gezalore** [comment]
I can get my hands on Cygwin, but not BSDs or OSX :neutral_face: 
tgmath is actually in verilated.cpp and you added it very recently:
d4f7f5297a0efca79bfa8f2aed6bb6b6be0206a3

**Comment by wsnyder** [comment]
Ah, think that was for a floating point hack I talked myself out of when realized it would probably bork any older compiler.  It should be OK to go.

**Comment by gezalore** [comment]
Travis now has FreeBSD: https://github.com/travis-ci/travis-yml/pull/151/files

**Comment by wsnyder** [comment]
I suggest file a bug, someone that uses free BSD regularly should maintain it.

File: include/verilated_vcd_sc.cpp [hunk 1]
```diff
@@ -17,7 +17,6 @@
 //=============================================================================
 // SPDIFF_OFF
 
-#include "verilatedos.h"
```
Comment (wsnyder) [inline include/verilated_vcd_sc.cpp:20]: I'd like to keep verilatedos.h first include in every file, even if later file includes something more specific.

**Comment by mjkillough** [comment]
FWIW, this PR still compiles for me on OpenBSD.

**Comment by gezalore** [comment]
Thanks @mjkillough that is really helpful.

**Comment by gezalore** [comment]
Cygwin confirmed OK too.

**Comment by madebr** [comment]
@wsnyder 

> These are certainly 95% cruft, but the question is which ones are the other 5% (or is it 0%)? tgmath.h is probably the most scary as that's quite specific & unusual.

Removing `tgmath.h` (and replacing it with `<cmath>`) lets Verilator compile with Visual Studio 2015:
see https://github.com/conan-io/conan-center-index/pull/1753#issuecomment-635986042

See https://en.cppreference.com/w/c/numeric/tgmath
It says there that `tgmath.h` is `math.h` + `complex.h` + some macros

See [the tgmath.h header of gcc](https://github.com/gcc-mirror/gcc/blob/41d6b10e96a1de98e90a7c0378437c3255814b16/gcc/ginclude/tgmath.h):

In c++ mode, `tgmath.h` is equivalent to `math.h`.

**Comment by wsnyder** [comment]
Can you make a pull request please (so Travis will check it). Thanks.

**Comment by madebr** [comment]
I replaced `tgmath.h` with `cmath` to let it compile.
I didn't check whether any of these headers were actually required.

If a next release fails to compile, I'll open a pr.

**Pull request merged**